src/load_gdm.cpp

```diff
@@ -0,0 +1,468 @@
+/*
+ * This source code is public domain.
+ *
+ * Authors: Alice Rowan <petrifiedrowan@gmail.com>
+*/
+
+////////////////////////////////////////////////////////////
+// General Digital Music module loader
+////////////////////////////////////////////////////////////
+
+#include "stdafx.h"
+#include "sndfile.h"
+
+static const DWORD GDM_SIG = 0xfe4d4447; // GDM\xFE
+static const DWORD GMFS_SIG = 0x53464d47; // GMFS
+
+typedef struct tagFILEHEADERGDM
```

Comment:
Paranoid: is byte packing guaranteed in these two structs?

Reply:
It should be—the GDM format isn't really friendly for struct packinh, which is why I made everything `char` and `BYTE` arrays. (If anything has a problem with this, it will almost certainly have problems with the unaligned structs everywhere else in libmodplug...)

Reply:
OK, thanks.